app/src/bcsc-theme/contexts/BCSCApiClientContext.tsx

```diff
@@ -29,42 +29,55 @@ export const BCSCApiClientContext = createContext<BCSCApiClientContextType | nul
  * @returns {*} {JSX.Element} The BCSCApiClientProvider component wrapping its children.
  */
 export const BCSCApiClientProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
-  const [store, dispatch] = useStore<BCState>()
+  const [store] = useStore<BCState>()
   const [isClientReady, setIsClientReady] = useState(false)
   const [error, setError] = useState<string | null>(null)
 
   const [logger] = useServices([TOKENS.UTIL_LOGGER])
 
-  const handleNewClient = (client: BCSCApiClient | null, errorMessage?: string) => {
+  const handleNewClient = useCallback((client: BCSCApiClient | null, errorMessage?: string) => {
     BCSC_API_CLIENT_SINGLETON = client
     setIsClientReady(Boolean(client))
     setError(errorMessage ?? null)
-  }
+  }, [])
 
   useEffect(() => {
     // Only attempt to configure the client if the store is loaded and the IAS API base URL is available
-    if (!store.stateLoaded || !store.developer.environment.iasApiBaseUrl) {
+    if (!store.stateLoaded || !store.developer.iasApiBaseUrl) {
       return
     }
 
     const configureClient = async () => {
-      setIsClientReady(false)
-      setError(null)
-
       let newClient = BCSC_API_CLIENT_SINGLETON
 
       try {
         // If the singleton doesn't exist or the base URL has changed, create a new instance
-        if (
-          !BCSC_API_CLIENT_SINGLETON ||
-          BCSC_API_CLIENT_SINGLETON.baseURL !== store.developer.environment.iasApiBaseUrl
-        ) {
-          newClient = new BCSCApiClient(store.developer.environment.iasApiBaseUrl, logger as RemoteLogger)
+        if (!BCSC_API_CLIENT_SINGLETON || BCSC_API_CLIENT_SINGLETON.baseURL !== store.developer.iasApiBaseUrl) {
+          // Only set isClientReady to false when we're actually changing clients
+          setIsClientReady(false)
+          setError(null)
+
+          // Clear old tokens when switching environments
+          if (BCSC_API_CLIENT_SINGLETON) {
+            BCSC_API_CLIENT_SINGLETON.tokens = undefined
+            BCSC_API_CLIENT_SINGLETON.tokensPromise = null
+          }
+
+          newClient = new BCSCApiClient(store.developer.iasApiBaseUrl, logger as RemoteLogger)
+
           await newClient.fetchEndpointsAndConfig()
 
           handleNewClient(newClient)
```

Comment:
nit: is this tokens reset needed?

When you instantiate the BCSCApiClient it will reset the tokens and `handleNewClient` will set the client singleton with the new client.